src/towncrier/test/test_check.py

```diff
@@ -121,3 +122,24 @@ def test_fragment_missing(self):
             self.assertTrue(
                 result.output.endswith("No new newsfragments found on this branch.\n")
             )
+
+    def test_none_stdout_encoding_works(self):
+        """
+        No failure when output is piped causing None encoding for stdout.
+        """
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            create_project("pyproject.toml")
+
+            fragment_path = "foo/newsfragments/1234.feature"
+            with open(fragment_path, "w") as f:
+                f.write("Adds gravity back")
+
+            call(["git", "add", fragment_path])
+            call(["git", "commit", "-m", "add a newsfragment"])
+
+            check_call(
+                [sys.executable, '-m', 'towncrier.check', "--compare-with", "master"],
+                stdout=PIPE,
+            )
```

Comment:
I don't like how this assertion is done.

From Python docs

Note
Do not use stdout=PIPE or stderr=PIPE with this function. The child process will block if it generates enough output to a pipe to fill up the OS pipe buffer as the pipes are not being read from. 

```
proc = subprocess.Popen([sys.executable, '-m', 'towncrier.check', "--compare-with", "master"], stdout=PIPE, stder=PIPE)
try:
    outs, errs = proc.communicate(timeout=15)
except TimeoutExpired:
    proc.kill()
    outs, errs = proc.communicate()
```

and then check that err is empty and the out has the expecte output and no sign of error.
Not sure how to check for no errors in stdout... but maybe in the future we can have towncrier sending all errors to stderr and then checking for an empty stderr is enough.

Reply:
Implemented without the timeout since py2 doesn't have timeout.  Could go for [`subprocess32`](https://pypi.org/project/subprocess32/) or something if it's particularly important  (sure, it's obviously nice).

Reply:
it's ok without the timeout